### PR TITLE
Don't package tests

### PR DESCRIPTION
**Pull request by jrbourbeau**
- [ ] Closes https://github.com/dask/distributed/issues/4902
- [ ] Tests added / passed
- [ ] Passes `black distributed` / `flake8 distributed` / `isort distributed`

Following-up on the conversation in https://github.com/dask/distributed/issues/4902, this PR modifies our `MANIFEST.in` to stop shipping tests. The resulting `diff` for the `SOURCE.txt` file generated by `python setup.py sdist bdist_wheel` is shown below.

cc @jakirkham @pentschev @fjetter 

<details>
<summary>SOURCE.txt diff:</summary>

```diff
72,76d71
< distributed/cli/tests/test_dask_scheduler.py
< distributed/cli/tests/test_dask_spec.py
< distributed/cli/tests/test_dask_ssh.py
< distributed/cli/tests/test_dask_worker.py
< distributed/cli/tests/test_tls_cli.py
86,90d80
< distributed/comm/tests/__init__.py
< distributed/comm/tests/test_comms.py
< distributed/comm/tests/test_ucx.py
< distributed/comm/tests/test_ucx_config.py
< distributed/comm/tests/test_ws.py
106,109d95
< distributed/dashboard/tests/test_bokeh.py
< distributed/dashboard/tests/test_components.py
< distributed/dashboard/tests/test_scheduler_bokeh.py
< distributed/dashboard/tests/test_worker_bokeh.py
120,128d105
< distributed/deploy/tests/test_adaptive.py
< distributed/deploy/tests/test_adaptive_core.py
< distributed/deploy/tests/test_cluster.py
< distributed/deploy/tests/test_deploy_utils.py
< distributed/deploy/tests/test_local.py
< distributed/deploy/tests/test_old_ssh.py
< distributed/deploy/tests/test_slow_adaptive.py
< distributed/deploy/tests/test_spec_cluster.py
< distributed/deploy/tests/test_ssh.py
139,148d115
< distributed/diagnostics/tests/test_eventstream.py
< distributed/diagnostics/tests/test_graph_layout.py
< distributed/diagnostics/tests/test_nvml.py
< distributed/diagnostics/tests/test_progress.py
< distributed/diagnostics/tests/test_progress_stream.py
< distributed/diagnostics/tests/test_progressbar.py
< distributed/diagnostics/tests/test_scheduler_plugin.py
< distributed/diagnostics/tests/test_task_stream.py
< distributed/diagnostics/tests/test_widgets.py
< distributed/diagnostics/tests/test_worker_plugin.py
163,164d129
< distributed/http/scheduler/tests/test_scheduler_http.py
< distributed/http/scheduler/tests/test_semaphore_http.py
191,193d155
< distributed/http/tests/__init__.py
< distributed/http/tests/test_core.py
< distributed/http/tests/test_routing.py
197d158
< distributed/http/worker/tests/test_worker_http.py
216,291d176
< distributed/protocol/tests/test_arrow.py
< distributed/protocol/tests/test_collection.py
< distributed/protocol/tests/test_collection_cuda.py
< distributed/protocol/tests/test_cupy.py
< distributed/protocol/tests/test_h5py.py
< distributed/protocol/tests/test_highlevelgraph.py
< distributed/protocol/tests/test_keras.py
< distributed/protocol/tests/test_netcdf4.py
< distributed/protocol/tests/test_numba.py
< distributed/protocol/tests/test_numpy.py
< distributed/protocol/tests/test_pandas.py
< distributed/protocol/tests/test_pickle.py
< distributed/protocol/tests/test_protocol.py
< distributed/protocol/tests/test_protocol_utils.py
< distributed/protocol/tests/test_rmm.py
< distributed/protocol/tests/test_scipy.py
< distributed/protocol/tests/test_serialize.py
< distributed/protocol/tests/test_sparse.py
< distributed/protocol/tests/test_torch.py
< distributed/tests/make_tls_certs.py
< distributed/tests/mytest.pyz
< distributed/tests/test_actor.py
< distributed/tests/test_as_completed.py
< distributed/tests/test_asyncprocess.py
< distributed/tests/test_batched.py
< distributed/tests/test_client.py
< distributed/tests/test_client_executor.py
< distributed/tests/test_client_loop.py
< distributed/tests/test_collections.py
< distributed/tests/test_config.py
< distributed/tests/test_core.py
< distributed/tests/test_counter.py
< distributed/tests/test_diskutils.py
< distributed/tests/test_events.py
< distributed/tests/test_failed_workers.py
< distributed/tests/test_ipython.py
< distributed/tests/test_locks.py
< distributed/tests/test_metrics.py
< distributed/tests/test_multi_locks.py
< distributed/tests/test_nanny.py
< distributed/tests/test_preload.py
< distributed/tests/test_priorities.py
< distributed/tests/test_profile.py
< distributed/tests/test_publish.py
< distributed/tests/test_pubsub.py
< distributed/tests/test_queues.py
< distributed/tests/test_resources.py
< distributed/tests/test_scheduler.py
< distributed/tests/test_security.py
< distributed/tests/test_semaphore.py
< distributed/tests/test_sizeof.py
< distributed/tests/test_spec.py
< distributed/tests/test_spill.py
< distributed/tests/test_steal.py
< distributed/tests/test_stress.py
< distributed/tests/test_system.py
< distributed/tests/test_system_monitor.py
< distributed/tests/test_threadpoolexecutor.py
< distributed/tests/test_tls_functional.py
< distributed/tests/test_utils.py
< distributed/tests/test_utils_comm.py
< distributed/tests/test_utils_perf.py
< distributed/tests/test_utils_test.py
< distributed/tests/test_variable.py
< distributed/tests/test_versions.py
< distributed/tests/test_worker.py
< distributed/tests/test_worker_client.py
< distributed/tests/testegg-1.0.0-py3.4.egg
< distributed/tests/tls-ca-cert.pem
< distributed/tests/tls-ca-key.pem
< distributed/tests/tls-cert-chain.pem
< distributed/tests/tls-cert.pem
< distributed/tests/tls-key-cert.pem
< distributed/tests/tls-key.pem
< distributed/tests/tls-self-signed-cert.pem
< distributed/tests/tls-self-signed-key.pem

```

</details>

### CODE REVIEW

**Review by fjetter — APPROVED** [review]

**Comment by mrocklin** [comment]
Checking in here

File: MANIFEST.in [hunk 1]
```diff
@@ -8,15 +8,13 @@ recursive-include distributed *.svg
 recursive-include distributed *.ico
 recursive-include distributed *.yaml
 recursive-include docs *.rst
+recursive-exclude distributed **/tests/*
 
 include setup.py
 include README.rst
 include LICENSE.txt
 include MANIFEST.in
 include requirements.txt
```
Comment (jakirkham) [inline MANIFEST.in:17]: There's a file called `pytest_resourceleaks.py`. Do we want to exclude that as well?

```suggestion
include requirements.txt
exclude distributed/pytest_resourceleaks.py
```

**Comment by jakirkham** [comment]
One question above. Otherwise LGTM. Generated an `sdist` and went through to make sure it seemed to be doing the right thing

Also would be great to have Peter's thoughts as well. Not sure if there were other things that he would like to see dropped

**Review by pentschev — APPROVED** [review]
From my side this looks good, I'm happy to have a well-defined policy regarding test shipping! 😄

**Comment by jrbourbeau** [comment]
Thanks for reviewing all, will merge once CI finishes

**Comment by jakirkham** [comment]
Thanks James! 😄

**Pull request merged**